From 7955d79a9ae233bb3367370bde12f6902bcbf2a7 Mon Sep 17 00:00:00 2001
From: Yanling Song <songyl@ramaxel.com>
Date: Wed, 23 Feb 2022 16:53:26 +0800
Subject: [PATCH] net/spnic: Fix array bounds error in ethtool
 get_link_ksettings
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 7955d79a9ae233bb3367370bde12f6902bcbf2a7
Modified-by-SEL: No


Ramaxel inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4TPKM
CVE: NA

-----------------------------------------------

Replace hw_link_mode by link_mode in ethtool get_link_ksettings

Signed-off-by: Yanling Song <songyl@ramaxel.com>
Reviewed-by: wenliang <wenliang@ramaxel.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/ramaxel/spnic/spnic_ethtool_stats.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool_stats.c b/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool_stats.c
index 8a5073a20f0b..5a830e3454d4 100644
--- a/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool_stats.c
+++ b/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool_stats.c
@@ -745,9 +745,9 @@ spnic_add_ethtool_link_mode(struct cmd_link_settings *link_settings, u32 hw_link
 	for (link_mode = 0; link_mode < LINK_MODE_MAX_NUMBERS; link_mode++) {
 		if (hw_link_mode & BIT(link_mode)) {
 			if (name == GET_SUPPORTED_MODE)
-				ETHTOOL_ADD_SUPPORTED_SPEED_LINK_MODE(link_settings, hw_link_mode);
+				ETHTOOL_ADD_SUPPORTED_SPEED_LINK_MODE(link_settings, link_mode);
 			else
-				ETHTOOL_ADD_ADVERTISED_SPEED_LINK_MODE(link_settings, hw_link_mode);
+				ETHTOOL_ADD_ADVERTISED_SPEED_LINK_MODE(link_settings, link_mode);
 		}
 	}
 }
-- 
2.26.2

